### ISSUE TITLE
Unmet assumptions don't mark test skipped/ignored

### ISSUE THREAD

**Issue by Philzen** [joel-costigliola/assertj-core#1679]
#### Summary

The [AssertJ docs](https://assertj.github.io/doc/#assertj-core-assumptions) do clearly state in section 3.15: 

> Assumptions provide support for conditional test execution, if the assumptions are met the test is executed normally, if they don’t the test is aborted and **marked as ignored**.

However, if i try to use Assumptions, the test in question always shows up as failed.

#### Example

```java
	OSType osType = null;
	assumeThat(osType)
		.withFailMessage("Could not determine whether OS is UNIX, MAC or WINDOWS")
		.isNotNull();    // ignore test if OS cannot be determined
```

But running `mvn test` always comes up with:

```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   OSConditionImplTest$ConditionIsNot.doesMatchThrowsExceptionIfCurrentOs:85 » AssumptionViolated
[INFO] 
[ERROR] Tests run: 260, Failures: 0, Errors: 1, Skipped: 0
```

In contrast, when using Junit 5 Assumptions instead:

```java
	OSType osType = null;
	Assumptions.assumeTrue(osType != null);
```

the result is as intended:

```
[INFO] Results:
[INFO] 
[WARNING] Tests run: 260, Failures: 0, Errors: 0, Skipped: 1
```

Not sure if i'm doing something wrong here... however from my understanding [AssertJ Assumptions were designed as a fully compatible replacement for Junit Assumptions](https://github.com/joel-costigliola/assertj-core/issues/975).

Tested with both versions 3.11.1 and 3.13.2

**Comment by joel-costigliola** [comment]
I have tried your code in a test and the test is properly skipped:
```
  @Test
  public void failed_assumption_should_lead_to_ignore_test() {
    // GIVEN
    String osType = null;
    assumeThat(osType).withFailMessage("Could not determine whether OS is UNIX, MAC or WINDOWS")
                      .isNotNull();
    // THEN the following assertion is not executed
    assertThat(false).isTrue();
  }
```
when executed, here's the output in eclipse:
![image](https://user-images.githubusercontent.com/382613/68064937-2bdfa480-fd87-11e9-8dad-c5382ab82b91.png)

Same thing when I run it from the command line:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.assertj.core.api.assumptions.Assumptions_assumeThat_Test
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.005 s - in org.assertj.core.api.assumptions.Assumptions_assumeThat_Test
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1
```
Could you share your environment or even better provide a reproducible test showing the behavior?

**Comment by PascalSchumacher** [comment]
I think AssertJ use the JUnit 4 assumption mechanism. Support for this was added in a later JUnit 5 version (could find the related release note entry, so I do not know the concrete version). Older JUnit 5 versions do not support it and display the behavior described in this issue.

@Philzen Please update your JUnit dependency to the latest version and everything should work as expected.

**Comment by Philzen** [comment]
Thanks so much for taking your valuable time off doing other awesome stuff (such as AssertJ) to help clarifying what i might be doing wrong here, much appreciated!

I'm running on a Manjaro linux machine

```
[phil@t520-laptop-me] 13:55 $ uname -a
Linux t520-laptop-me 4.19.80-1-MANJARO #1 SMP PREEMPT Fri Oct 18 12:28:02 UTC 2019 x86_64 GNU/Linux
```

The project i'm using has not direct junit 4 reference whatsoever, here's the `pom.xml` of the project i'm working on:

```xml
<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>de.*****.**</groupId>
	<artifactId>******</artifactId>
	<version>4.0.1.520</version>
	<description>********</description>

	<properties>
		<java.version>1.8</java.version>
		<maven.compiler.source>${java.version}</maven.compiler.source>
		<maven.compiler.target>${java.version}</maven.compiler.target>
		<springfox-version>2.7.0</springfox-version>
		<springfox-version>2.7.0</springfox-version>
		<validation-version>2.0.1.Final</validation-version>
		<docker.image.prefix>***.*******.***:***</docker.image.prefix>
	</properties>

	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.1.1.RELEASE</version>
	</parent>

	<build>
		<sourceDirectory>src/main/java</sourceDirectory>
		<testSourceDirectory>src/test/java</testSourceDirectory>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-jar-plugin</artifactId>
				<version>3.1.1</version>
				<executions>
					<execution>
						<phase>package</phase>
						<goals>
							<goal>jar</goal>
						</goals>
						<configuration>
							<outputDirectory>${project.basedir}/src/main/resources</outputDirectory>
							<finalName>stream</finalName>
							<classifier>engine</classifier>
							<includes>
								<include>**/engine/**</include>
								<include>**/de/moovit/mcc/**</include>
							</includes>
						</configuration>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<groupId>org.codehaus.mojo</groupId>
				<artifactId>build-helper-maven-plugin</artifactId>
				<executions>
					<execution>
						<id>add-source</id>
						<phase>generate-sources</phase>
						<goals>
							<goal>add-source</goal>
						</goals>
						<configuration>
							<sources>
								<source>${project.basedir}/target/generated/src/main/java</source>
							</sources>
						</configuration>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<groupId>org.openapitools</groupId>
				<artifactId>openapi-generator-maven-plugin</artifactId>
				<version>4.0.0-beta</version>
				<executions>
					<execution>
						<goals>
							<goal>generate</goal>
						</goals>
						<configuration>
							<inputSpec>${project.basedir}/src/main/resources/api.yaml</inputSpec>
							<generatorName>spring</generatorName>
							<output>target/generated</output>
							<validateSpec>false</validateSpec>
							<configOptions>
								<interfaceOnly>true</interfaceOnly>
								<sourceFolder>src/main/java</sourceFolder>
								<apiPackage>${project.groupId}.api</apiPackage>
								<modelPackage>${project.groupId}.model</modelPackage>
								<invokerPackage>${project.groupId}.invoker</invokerPackage>
							</configOptions>
							<ignoreFileOverride>${project.basedir}/.openapi-codegen-ignore</ignoreFileOverride>
						</configuration>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
				<executions>
					<execution>
						<goals>
							<goal>repackage</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<groupId>com.spotify</groupId>
				<artifactId>dockerfile-maven-plugin</artifactId>
				<version>1.4.10</version>
				<executions>
					<execution>
						<id>default</id>
						<goals>
							<goal>build</goal>
						</goals>
					</execution>
				</executions>
				<dependencies>
					<dependency>
						<groupId>com.spotify</groupId>
						<artifactId>docker-client</artifactId>
						<version>8.11.7</version>
					</dependency>
				</dependencies>
				<configuration>
					<username>admin</username>
					<password>bitte0479</password>
					<repository>${docker.image.prefix}/mcc_${project.name}</repository>
					<tag>${project.version}</tag>

					<buildArgs>
						<JAR_FILE>${project.build.finalName}.jar</JAR_FILE>
					</buildArgs>
				</configuration>
			</plugin>
			<plugin>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>2.22.0</version>
			</plugin>
		</plugins>
	</build>

	<dependencies>
		<dependency>
			<groupId>org.openapitools</groupId>
			<artifactId>openapi-generator</artifactId>
			<version>4.0.0-beta</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-mongodb</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-amqp</artifactId>
		</dependency>

		<dependency>
			<groupId>io.springfox</groupId>
			<artifactId>springfox-swagger2</artifactId>
			<version>${springfox-version}</version>
		</dependency>

		<dependency>
			<groupId>io.springfox</groupId>
			<artifactId>springfox-swagger-ui</artifactId>
			<version>${springfox-version}</version>
		</dependency>

		<dependency>
			<groupId>com.atlassian.commonmark</groupId>
			<artifactId>commonmark</artifactId>
			<version>0.12.1</version>
		</dependency>

		<dependency>
			<groupId>joda-time</groupId>
			<artifactId>joda-time</artifactId>
		</dependency>

		<dependency>
			<groupId>org.openapitools</groupId>
			<artifactId>jackson-databind-nullable</artifactId>
			<version>0.1.0</version>
		</dependency>

		<dependency>
			<groupId>com.google.code.gson</groupId>
			<artifactId>gson</artifactId>
			<version>2.8.1</version>
		</dependency>

		<dependency>
			<groupId>org.apache.commons</groupId>
			<artifactId>commons-exec</artifactId>
			<version>1.3</version>
		</dependency>
		<dependency>
			<groupId>javax.ws.rs</groupId>
			<artifactId>javax.ws.rs-api</artifactId>
			<version>2.1.1</version>
		</dependency>
		<dependency>
			<groupId>org.glassfish.jersey.core</groupId>
			<artifactId>jersey-client</artifactId>
			<version>2.28</version>
		</dependency>

		<dependency>
			<groupId>com.danielflower.apprunner</groupId>
			<artifactId>javasysmon</artifactId>
			<version>0.3.5.1</version>
		</dependency>

		<dependency>
			<groupId>com.sun.mail</groupId>
			<artifactId>smtp</artifactId>
			<version>1.4.5</version>
		</dependency>

		<dependency>
			<groupId>org.junit.jupiter</groupId>
			<artifactId>junit-jupiter</artifactId>
			<version>5.5.2</version>
			<scope>test</scope>
		</dependency>

		<dependency>
			<groupId>org.assertj</groupId>
			<artifactId>assertj-core</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>
	<name>component</name>
</project>
```

When i click on the assertj-core artifactId and let IntelliJ bring up the `assertj-core-3.11.1.pom` that maven pulled from the repository, we can see a junit4-dependency in there indeed (however is also includes junit5-jupiter, which also brings `org.junit.jupiter.api.Assumptions` that i use in my example above that works as expected):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <parent>
    <artifactId>assertj-parent-pom</artifactId>
    <groupId>org.assertj</groupId>
    <version>2.2.1</version>
    <relativePath>pom.xml</relativePath>
  </parent>
  <modelVersion>4.0.0</modelVersion>
  <artifactId>assertj-core</artifactId>
  <packaging>bundle</packaging>
  <name>AssertJ fluent assertions</name>
  <version>3.11.1</version>
  <description>Rich and fluent assertions for testing for Java</description>
  <issueManagement>
    <system>github</system>
    <url>https://github.com/joel-costigliola/assertj-core/issues</url>
  </issueManagement>
  <inceptionYear>2014</inceptionYear>
  <mailingLists>
    <mailingList>
      <name>AssertJ Group</name>
      <subscribe>http://groups.google.com/group/assertj</subscribe>
      <unsubscribe>http://groups.google.com/group/assertj</unsubscribe>
      <post>http://groups.google.com/group/assertj</post>
    </mailingList>
  </mailingLists>
  <scm>
    <connection>scm:git:git@github.com:joel-costigliola/assertj-core.git</connection>
    <developerConnection>scm:git:git@github.com:joel-costigliola/assertj-core.git</developerConnection>
    <tag>assertj-core-3.11.1</tag>
    <url>git@github.com:joel-costigliola/assertj-core</url>
  </scm>
  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <groupId>org.eclipse.m2e</groupId>
          <artifactId>lifecycle-mapping</artifactId>
          <version>1.0.0</version>
          <configuration>
            <lifecycleMappingMetadata>
              <pluginExecutions>
                <pluginExecution>
                  <pluginExecutionFilter>
                    <groupId>com.mycila</groupId>
                    <artifactId>license-maven-plugin</artifactId>
                    <versionRange>[2.6,)</versionRange>
                    <goals>
                      <goal>format</goal>
                    </goals>
                  </pluginExecutionFilter>
                  <action>
                    <ignore />
                  </action>
                </pluginExecution>
              </pluginExecutions>
            </lifecycleMappingMetadata>
          </configuration>
        </plugin>
      </plugins>
    </pluginManagement>
    <plugins>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <configuration>
          <source>1.8</source>
          <target>1.8</target>
          <encoding>${project.build.sourceEncoding}</encoding>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-jar-plugin</artifactId>
        <configuration>
          <archive>
            <manifestEntries>
              <Automatic-Module-Name>org.assertj.core</Automatic-Module-Name>
            </manifestEntries>
          </archive>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>2.22.0</version>
        <configuration>
          <argLine>${argLine}</argLine>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-shade-plugin</artifactId>
        <version>3.1.1</version>
        <executions>
          <execution>
            <phase>package</phase>
            <goals>
              <goal>shade</goal>
            </goals>
            <configuration>
              <createDependencyReducedPom>true</createDependencyReducedPom>
              <relocations>
                <relocation>
                  <pattern>net.bytebuddy</pattern>
                  <shadedPattern>org.assertj.core.internal.bytebuddy</shadedPattern>
                </relocation>
              </relocations>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-dependency-plugin</artifactId>
        <executions>
          <execution>
            <id>unpack-shade</id>
            <phase>package</phase>
            <goals>
              <goal>unpack</goal>
            </goals>
            <configuration>
              <artifactItems>
                <artifactItem>
                  <groupId>${project.groupId}</groupId>
                  <artifactId>${project.artifactId}</artifactId>
                  <version>${project.version}</version>
                </artifactItem>
              </artifactItems>
              <outputDirectory>${project.build.directory}/classes</outputDirectory>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <groupId>org.apache.felix</groupId>
        <artifactId>maven-bundle-plugin</artifactId>
        <version>3.4.0</version>
        <extensions>true</extensions>
        <executions>
          <execution>
            <id>bundle-after-unpack-shaded</id>
            <phase>package</phase>
            <goals>
              <goal>bundle</goal>
            </goals>
            <configuration>
              <instructions>
                <Export-Package>org.assertj.core.*</Export-Package>
                <Import-Package>!org.assertj.core.*,
                  org.junit;resolution:="optional",
                  org.opentest4j;resolution:="optional",
                  org.testng;resolution:="optional",
                  *</Import-Package>
                <Bundle-RequiredExecutionEnvironment>JavaSE-1.8</Bundle-RequiredExecutionEnvironment>
                <_removeheaders>Bnd-LastModified</_removeheaders>
              </instructions>
              <niceManifest>true</niceManifest>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <groupId>org.jacoco</groupId>
        <artifactId>jacoco-maven-plugin</artifactId>
        <executions>
          <execution>
            <id>jacoco-report</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>report</goal>
            </goals>
          </execution>
          <execution>
            <id>default-check</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>check</goal>
            </goals>
            <configuration>
              <rules>
                <rule>
                  <element>BUNDLE</element>
                  <limits>
                    <limit>
                      <counter>CLASS</counter>
                      <value>COVEREDRATIO</value>
                      <minimum>0.98</minimum>
                    </limit>
                  </limits>
                </rule>
              </rules>
            </configuration>
          </execution>
        </executions>
        <configuration>
          <excludes>
            <exclude>**/*hamcrest*/**</exclude>
          </excludes>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-javadoc-plugin</artifactId>
        <configuration>
          <stylesheetfile>src/main/javadoc/assertj-javadoc.css</stylesheetfile>
          <top>&lt;script src="http://cdn.jsdelivr.net/highlight.js/8.6/highlight.min.js"&gt;&lt;/script&gt;</top>
          <additionalOptions>--allow-script-in-comments</additionalOptions>
          <footer>&lt;script type="text/javascript"&gt;
              hljs.initHighlightingOnLoad();
            &lt;/script&gt;</footer>
        </configuration>
      </plugin>
    </plugins>
  </build>
  <profiles>
    <profile>
      <id>java9+</id>
      <properties>
        <argLine>--add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED
          --add-opens java.base/java.io=ALL-UNNAMED --add-opens java.base/java.math=ALL-UNNAMED</argLine>
      </properties>
    </profile>
    <profile>
      <id>java11+</id>
      <properties>
        <sonar.skip>true</sonar.skip>
        <jacoco.skip>true</jacoco.skip>
      </properties>
    </profile>
  </profiles>
  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>4.12</version>
      <scope>provided</scope>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>org.junit.platform</groupId>
      <artifactId>junit-platform-runner</artifactId>
      <version>1.2.0</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>apiguardian-api</artifactId>
          <groupId>org.apiguardian</groupId>
        </exclusion>
        <exclusion>
          <artifactId>junit-platform-launcher</artifactId>
          <groupId>org.junit.platform</groupId>
        </exclusion>
        <exclusion>
          <artifactId>junit-platform-suite-api</artifactId>
          <groupId>org.junit.platform</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.junit.vintage</groupId>
      <artifactId>junit-vintage-engine</artifactId>
      <version>5.2.0</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>junit-platform-engine</artifactId>
          <groupId>org.junit.platform</groupId>
        </exclusion>
        <exclusion>
          <artifactId>apiguardian-api</artifactId>
          <groupId>org.apiguardian</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-api</artifactId>
      <version>5.2.0</version>
      <scope>provided</scope>
      <exclusions>
        <exclusion>
          <artifactId>junit-platform-commons</artifactId>
          <groupId>org.junit.platform</groupId>
        </exclusion>
        <exclusion>
          <artifactId>apiguardian-api</artifactId>
          <groupId>org.apiguardian</groupId>
        </exclusion>
      </exclusions>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-engine</artifactId>
      <version>5.2.0</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>junit-platform-engine</artifactId>
          <groupId>org.junit.platform</groupId>
        </exclusion>
        <exclusion>
          <artifactId>apiguardian-api</artifactId>
          <groupId>org.apiguardian</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter-params</artifactId>
      <version>5.2.0</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>apiguardian-api</artifactId>
          <groupId>org.apiguardian</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>org.opentest4j</groupId>
      <artifactId>opentest4j</artifactId>
      <version>1.1.0</version>
      <scope>provided</scope>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>org.hamcrest</groupId>
      <artifactId>hamcrest-core</artifactId>
      <version>1.3</version>
      <scope>provided</scope>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>org.mockito</groupId>
      <artifactId>mockito-core</artifactId>
      <version>2.20.1</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>byte-buddy-agent</artifactId>
          <groupId>net.bytebuddy</groupId>
        </exclusion>
        <exclusion>
          <artifactId>objenesis</artifactId>
          <groupId>org.objenesis</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>com.github.marschall</groupId>
      <artifactId>memoryfilesystem</artifactId>
      <version>1.2.1</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>com.google.guava</groupId>
      <artifactId>guava</artifactId>
      <version>25.0-jre</version>
      <scope>test</scope>
      <exclusions>
        <exclusion>
          <artifactId>jsr305</artifactId>
          <groupId>com.google.code.findbugs</groupId>
        </exclusion>
        <exclusion>
          <artifactId>checker-compat-qual</artifactId>
          <groupId>org.checkerframework</groupId>
        </exclusion>
        <exclusion>
          <artifactId>error_prone_annotations</artifactId>
          <groupId>com.google.errorprone</groupId>
        </exclusion>
        <exclusion>
          <artifactId>j2objc-annotations</artifactId>
          <groupId>com.google.j2objc</groupId>
        </exclusion>
        <exclusion>
          <artifactId>animal-sniffer-annotations</artifactId>
          <groupId>org.codehaus.mojo</groupId>
        </exclusion>
      </exclusions>
    </dependency>
  </dependencies>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.junit</groupId>
        <artifactId>junit-bom</artifactId>
        <version>5.2.0</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
</project>
```

Kindly advice if i can provide any additional info to help clarify the root cause of my issue.

**Comment by PascalSchumacher** [comment]
You are using JUnit `5.2.0`. This version does not support the JUnit 4 assumption mechanism used by AssertJ. Try JUnit `5.5.2` and it should work.

**Comment by Philzen** [comment]
> You are using JUnit `5.2.0`. This version does not support the JUnit 4 assumption mechanism used by AssertJ. Try JUnit `5.5.2` and it should work.

Pardon me, is it possible you are confusing the two POMs that i posted? The project i'm working on explicitely uses junit 5.5.2 in it's POM - it's the AssertJ 3.11.1 pom that includes a junit 5.2.0 reference.

However, when using assertJ v3.13.2 (which has jUnit 5.5.1) - it's still the same problem here.

**Comment by scordio** [comment]
Could you also try to upgrade AssertJ to 3.14.0 which uses JUnit 5.5.2?

**Comment by PascalSchumacher** [comment]
> Pardon me, is it possible you are confusing the two POMs that i posted? The project i'm working on explicitely uses junit 5.5.2 in it's POM - it's the AssertJ 3.11.1 pom that includes a junit 5.2.0 reference.

Yes, sorry.

I see you are using Spring Boot `2.1.1` whose dependency management fixes the JUnit 5 dependencies to version `5.3.2`. You declare a junit-jupiter `5.5.2` dependency, but this artifact just aggregates/adds the `junit-jupiter-api`, `junit-jupiter-params`, `junit-jupiter-engine` dependencies. The version of these dependencies is changed to `5.3.2` by Spring Boot. So in effect you are using JUnit `5.3.2`.

You can validate this by checking the `mvn dependency:tree` output.

`spring-boot-starter-test` add JUnit 4 to the classpath and therefore AssertJ use the JUnit 4 assumption mechanism (throwing `org.junit.AssumptionViolatedException`). If JUnit 4 is not on the classpath recent version of AssertJ use the JUnit 5 assumption mechanism (throwing `org.opentest4j.TestAbortedException`).

So ensure that JUnit `5.5.2` is used (or exclude the JUnit 4 dependency if you do not need it).

**Comment by Philzen** [comment]
This comment as reference for anybody else coming to this issue looking for a simple fix:

~a year later and i finally got what you were saying. Everybody in this thread: a **big thank you** for having taken your valuable time to look into this. I guess back then i was too afraid to waste it any more and just left things the way they were, but having returned to the project this year i took the time to understand Effective POMs and dependency trees a bit better, so i could make out what is necessary (at least until i get the product team to upgrade spring boot to 2.2, which upgrades JUnit to 5.5.x).

One simply needs to override the `<junit-jupiter.version>`, which shadows the version pulled in from spring boot in the dependency tree, so it's:

```xml
	<properties>
		...
		<!-- upgrade parent (spring boot) dependency from 5.3.x  -->
		<!-- can be removed when spring boot is upgraded to >= 2.2 -->
		<junit-jupiter.version>5.5.2</junit-jupiter.version>
	</properties>
```

That does the trick. BTW - in the meanwhile i started using the same mechanism to ensure i get the latest version of assertJ pulled in (which also helps keeping the `<dependencies>`-section shorter) by simply adding `<assertj.version>3.19.0</assertj.version>`  in the same `<properties>`-section.
